File: src/main/java/me/senseiwells/arucas/api/docs/FunctionDoc.java
package me.senseiwells.arucas.api.docs;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface FunctionDoc {
	boolean isVarArgs() default false;
	boolean isStatic() default false;
	String[] deprecated() default { };
	String name();
	String[] desc();
	String[] params() default { };
	String[] returns() default { };
	String[] throwMsgs() default { };
	String[] example();
}

File: src/main/kotlin/me/senseiwells/arucas/api/docs/FunctionDoc.kt
package me.senseiwells.arucas.api.docs

@Deprecated("me.senseiwells.arucas.api.docs.annotations.FunctionDoc should be used instead.")
@Target(AnnotationTarget.FUNCTION)
annotation class FunctionDoc(
    val isVarArgs: Boolean = false,
    val isStatic: Boolean = false,
    val deprecated: Array<String> = [],
    val name: String,
    val desc: Array<String>,
    val params: Array<String> = [],
    val returns: Array<String> = [],
    @Deprecated("This field will no longer be included")
    val throwMsgs: Array<String> = [],
    val examples: Array<String>
)